### DISCUSSION TITLE
More aggressive discharging after new prices

### DISCUSSION THREAD

**Discussion by doncuco** [sonnen-tools/node-red#17] [Ideas]
Hi,
my battery with 22kwh was loaded to nearly 95% last night. and right now it is only discharged to 70%. 
So at 13:00 the new prices arrived. And they are way cheaper than before. 

We know the price for discharging. 
We know the prices from 13:00 till we get the first window to load. And we know that we load Batterysize/loadPower = hours to load

so very simple: if (average Tibber price till next charging window - batt discharge price) + (average Tibber price till next charging window - new discharge price of battery) > 0
The 0 could also be a variable set to 1 or 2 cent. 

then it should be better to use the energy from the battery. 
So how do we calculate then?
We need to check if we have discharge windows already set? How many?
Do we know the average consumption of the house of the last hours?
if yes then we could use the remaining capacity of the battery and divide it through that.
or you could use the max discharging value, which seems to high in my opinion. In that case I would create a variable for that.

and then calculate the amount of hours we can use the battery, before reloading starts.
We Substrat the discharge window hours and if the value is bigger then 0, we search the next most expensive hours.

Just an idea. What do you think about?
If you need more information to the functions and calculation, then let me know.

Best regards.

Tobias

**Comment by markusjnagel** [comment]
Yes, let me think about that - seems like a good idea. Also (if you haven't seen it already) I'm currently working on something to accommodate the high price fluctuations currently in northern Europe (checki the other discussions).
